### PR TITLE
Add Controller Request specs

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;

- Added RSpec as a gem to the project.

- Created a Request spec file for all your controllers with the following checks.
`If response status was correct.`
`If a correct template was rendered.`
`If the response body includes correct placeholder text.`

### CODE REVIEW

File: spec/requests/posts_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  it 'should get show' do
+    get "/users#{user.id}/posts#{post.id}"
+    expect(response).to have_http_status(200)
+  end
+
+  it 'returns correct show template' do
+    get "/users#{user.id}/posts#{post.id}"
+    expect(response).to render_template(:show)
+  end
```
Comment (Munsa1) [inline spec/requests/posts_request_spec.rb:20]: - [ ] Kindly fix your tests, all of them are failing, now I dont know if this is a configuration issue, but when I run them, I get the following result.
image

## Note that if it is a configuration issue, leave a comment for the next reviewer

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @JonahKayizzi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_request_spec.rb (hunk 1 above)
Comment (JonahKayizzi) [reply spec/requests/posts_request_spec.rb:20]: Hello @Munsa1, this feedback is not sufficient and provides no guidance on a way forward.
The tests are all successful on my side
image

**Review by addod19 — APPROVED** [review]
### Hi @JonahKayizzi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Although I had the same issue with the previous code reviewer, once I saw you systems passed all the 25 test cases, I realized it could be platform or OS issue

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**